projects/demo/src/modules/charts/line-days-chart/examples/1/index.ts

```diff
@@ -27,19 +29,22 @@ export class TuiLineDaysChartExample1 {
 
     readonly maxLength: TuiDayLike = {month: 12};
 
-    constructor(@Inject(TUI_MONTHS) private readonly months: readonly string[]) {}
+    readonly xStringify$: Observable<TuiStringHandler<TuiDay>> = this.months$.pipe(
+        map(months => {
+            return ({month, day}) => `${months[month]}, ${day}`;
+        }),
+    );
+
+    constructor(@Inject(TUI_MONTHS) readonly months$: Observable<readonly string[]>) {}
 
-    get labels(): readonly string[] {
-        return this.computeLabels(this.range, this.months);
+    get labels$(): Observable<readonly string[]> {
```

Comment:
This creates a new `Observable` on every change detection cycle. `computeLabels` is a function with memoization (it uses `@tuiPure` decorator) so you need to move `Observable` creation there to get rid of unnecessary recalculations and improve performance.

Reply:
Check it out again, please